### PR TITLE
feat(vue 3): support vue 3

### PR DESCRIPTION
**Pull request by eunjae-lee**
## Summary

This is a base branch for all the Vue 3 related PRs.

## How to try this branch

1. scroll to bottom
2. click the status icon of the latest commit
3. click "Details" on "ci/codesandbox" item
4. get the url to install the temporary version from this branch
5. import like this:

```js
import InstantSearch from 'vue-instantsearch/vue3/es'
```

6. let us know how it goes and if you see any bug

https://user-images.githubusercontent.com/499898/124242072-a6446700-db1c-11eb-987a-6c0e5d86b8a5.mp4

## Major TODOs

- [x] vue-router@4: It just works, only with one change on the userland: `vueRouter.currentRoute.query` → `vueRouter.currentRoute.value.query` (need update the doc and the example)
- [x] Vue 3 with vue-cli
- [x] Vue 3 with vite
- [x] Vue 3 SSR with vue-cli
- [x] Vue 3 SSR with vite
- [ ] Nuxt 3 (when it becomes public beta, which isn't yet)

### CODE REVIEW

**Comment by Kocal** [comment]
Nice job @eunjae-lee! :sparkles: :heart: 
image

**Comment by Haroenv** [comment]
let's make sure we rebase instead of squashing this PR to preserve the history

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,60 @@
+# [4.0.0-beta.3](https://github.com/algolia/vue-instantsearch/compare/v4.0.0-beta.2...v4.0.0-beta.3) (2021-08-20)
```
Comment (Haroenv) [inline CHANGELOG.md:1]: should we change this to just say v4 and have all changes to v3?

Comment (eunjae-lee) [reply CHANGELOG.md:1]: Yes, I'm planning to update it in a release PR for 4.0.0.

**Review by Haroenv — APPROVED** [review]

**Pull request merged**